packages/Input/src/Input.js

```diff
@@ -4,25 +4,76 @@ import classNames from "classnames";
 import * as constants from "@paprika/constants/lib/Constants";
 import TimesCircleIcon from "@paprika/icon/lib/TimesCircle";
 import Button from "@paprika/button";
+import useI18n from "@paprika/l10n/lib/useI18n";
 import * as types from "./types";
 import * as sc from "./Input.styles";
 
+function callAll(...fns) {
+  return (...args) => {
+    for (const fn of fns) {
+      if (fn) fn(...args);
+    }
+  };
+}
+
 const Input = React.forwardRef((props, ref) => {
-  const inputClearHandler = e => {
-    e.target.value = "";
-    props.onChange(e);
```

Comment:
This is potentially a breaking change, but it only worked in some situations by accident.  The `e` in this case is the `click` event on the Clear Button, not the `change` event on the `<input>` so if the consuming app sets the state based on the `e` passed to `onChange` then the `target.value` is undefined and the input will be cleared.  

This is flawed.

Going forward the consumer should also connect the `onClear` callback to a handler that will clear the value (if the component is controlled).

Reply:
🤔 I don't think this is completely wrong, this has good abstraction, the end-user doesn't need to know that the onClear Button even exist, Possible solution is to manually dispatch the event.

```
const event = new Event('change');

// Dispatch it.
refInput.current.dispatchEvent(event);
```

I think that would be a nicer solution than forcing a developer to wire the onClear callback.

Reply:
I actually tried this. It did not actually trigger the `handleChange` or `onChange` callbacks though 😒 

Maybe I was doing something wrong, but yeah, I tried `bestRef.current.dispatchEvent(new Event('change');` and nothing would actually happen.

Reply:
`onChange({ target: { value: "" }})` 🙈

Reply:
That will work in a very specific use, when the consumer only cares about `event.target.value`... do you think that's acceptable?

Reply:
jokes aside if you can make it work with creating the event would be a way fancier and better experience, what you can do is to create an `id={`input_uuidV4`}` and attach it to the input, then with javascript find the real dom element, and use tha one to fire the Event, I'm suspecting that react doesn't work because we are trying to mount on top of the synthetic event.

Reply:
I also tried:

```js
const event = document.createEvent("HTMLEvents");
event.initEvent("change", true, false);
document.getElementById("tester").dispatchEvent(event);
```

From https://stackoverflow.com/questions/39065010/why-react-event-handler-is-not-called-on-dispatchevent

But still no luck.  Looked at this package: https://github.com/vitalyq/react-trigger-change

But it says clearly in the README
> This module is a hack and is tightly coupled with React's implementation details. Not intended for production use. Useful for end-to-end testing and debugging.

It seems like this kind of thing is deliberately not supported by React.

Reply:
what would happens if we trigger the onchange with a null no with the event, would that be too bad?
```
handleChange(event) {
  if (event) set(event.target.value) : set("")
}
```
this will work as well in ts signature because you could say is an event type | null

Reply:
Yeah, I can dig that. But then I think we should remove the `onClear` prop, which is definitely a breaking change.

Reply:
Also, we should match this behaviour in the `<ListBox>`, but that's a separate task 😉

Reply:
unless some bug was introduced the onclear should report an onChange as well in the listbox...

Reply:
yeah I think is a breaking change...